Tasks/Common/nuget-task-common/NuGetToolGetter.ts

```diff
@@ -0,0 +1,92 @@
+import * as toolLib from 'vsts-task-tool-lib/tool';
+import * as taskLib from 'vsts-task-lib/task';
+import * as restm from 'typed-rest-client/RestClient';
+
+interface INuGetTools {
+    nugetexe: INuGetVersionInfo[]
+}
+
+interface INuGetVersionInfo {
+    version: string,
+    url: string,
+    stage: NuGetReleaseStage
+}
+
+enum NuGetReleaseStage
+{
+    "EarlyAccessPreview",
+    "Released",
+    "ReleasedAndBlessed"
+}
+
+export async function getNuGet(versionSpec: string) {
+    toolLib.debug('Trying to get tool from local cache');
+    let localVersions: string[] = toolLib.findLocalToolVersions('NuGet');
+    let version: string = toolLib.evaluateVersions(localVersions, versionSpec);
+
+    if (toolLib.isExplicitVersion(versionSpec)) {
+        toolLib.debug('Exact match expected on version: ' + versionSpec);
+    }
+    else {
+        toolLib.debug('Query match expected on version: ' + versionSpec);
+        console.log(taskLib.loc("Info_ExpectBehaviorChangeWhenUsingVersionQuery"));
+    }
+
+    if (version) {
+        console.log(taskLib.loc("Info_ResolvedToolFromCache", version));
+    }
+    else {
+        toolLib.debug('Querying versions list');
+        
+        let versionsUrl = 'https://dist.nuget.org/tools.json';
+        let rest: restm.RestClient = new restm.RestClient('vsts-tasks/NuGetToolInstaller');
+        
+        let nugetVersions: INuGetVersionInfo[] = (await rest.get<INuGetVersionInfo[]>(versionsUrl, GetRestClientOptions())).result;
+        let availableVersions: INuGetVersionInfo[] = nugetVersions.filter(x => x.stage !== NuGetReleaseStage.EarlyAccessPreview);
+        let filteredVersions: string[] = availableVersions.map(x => x.version);
+        let versionUrlMap: Map<string, string> = new Map<string, string>();
+        availableVersions.forEach((versionInfo:INuGetVersionInfo) => {
+            versionUrlMap[versionInfo.version.toLowerCase()] = versionInfo.url;
+        });
+
+        version = toolLib.evaluateVersions(filteredVersions, versionSpec);
+        if(!version)
+        {
+            toolLib.debug('No version was found which matches ' + versionSpec);
```

Comment:
we can use taskLib.debug for all logging. Minor, just looks funny. toolLib.debug just calls in to taskLib.debug.